src/Uno.UI/UI/Xaml/UIElement.wasm.cs

```diff
@@ -175,7 +175,15 @@ protected internal void SetClasses(string[] cssClasses, int index = -1)
 		/// <param name="clipRect">The Clip rect to set, if any</param>
 		protected internal void ArrangeElementNative(Rect rect, bool clipToBounds, Rect? clipRect)
 		{
-			LayoutSlotWithMarginsAndAlignments = rect;
+			var parent = VisualTreeHelper.GetParent(this) as UIElement;
```

Comment:
I think that we are be able to the parent's padding more easily here. This method is invoked by the parent so at least we should just give it as a parameter.

Reply:
Can't be optimized.  A light refactoring has been done.